File: Semester 1/Class13_Computer_Contest_Lv_2_Notes_and_Homework_Dec_04-05/lvl2_hw_sol/lvl2_a_familiar_problem.cpp
#include <bits/stdc++.h>
using namespace std;
typedef long long ll;
int main(){
    int n; ll m;
    cin >> n >> m;
    ll psa[n+1]={};
    for(int i=1; i<=n; i++){
        cin >> psa[i];
        psa[i] += psa[i-1];
    }
    int ans = 0;
    for(int l=1, r=1; r<=n; r++){
        while(psa[r] - psa[l-1] >= m){
            l++;
        }
        ans = max(ans, r - l + 1);
    }
    cout << ans << endl;
}

File: Semester 1/Class13_Computer_Contest_Lv_2_Notes_and_Homework_Dec_04-05/lvl2_hw_sol/lvl2_a_familiar_problem.py
import sys
input = sys.stdin.readline
n, m = map(int, input().split())
line = list(map(int, input().split()))
psa = [0]
for i in range(1, len(line) + 1):
    psa.append(line[i - 1] + psa[i - 1])
l, r, ans = 1, 1, 0
while r <= n:
    while psa[r] - psa[l-1] >= m :  # sum in [l, r] >= m
        l += 1
    ans = max(ans, r - l + 1)   # once get out of the loop -> sum [l, r] < m
    r += 1
print(ans)